### PR TITLE
Add jupyterlab s3 browser

### PR DESCRIPTION
**Pull request by janjagusch**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/singleton-decorator/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/Kemaweyan/singleton_decorator
+  summary: A testable singleton decorator
+  license: AGPL-3.0-or-later
+  license_file: PLEASE_ADD_LICENSE_FILE
```
Comment (janjagusch) [inline recipes/singleton-decorator/meta.yaml:36]:
```suggestion
```

**Review by martinRenou — APPROVED** [review]
LGTM, thanks!

**Comment by janjagusch** [comment]
@conda-forge/python, please take a look.

File: recipes/singleton-decorator/LICENSE [hunk 2]
```diff
@@ -0,0 +1,219 @@
+GNU GENERAL PUBLIC LICENSE
```
Comment (xhochy) [inline recipes/singleton-decorator/LICENSE:1]: The license mentioned in the metadata is not the same. This is GPL whereas the recipe states AGPL.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/singleton-decorator/LICENSE (hunk 2 above)
Comment (janjagusch) [reply recipes/singleton-decorator/LICENSE]: Good catch. I have updated the license.

**Comment by janjagusch** [comment]
Summary of what's blocking this PR:
- [jupyterlab-s3-browser](https://github.com/IBM/jupyterlab-s3-browser) depends on [singleton_decorator](https://github.com/Kemaweyan/singleton_decorator), [which is missing a license](https://github.com/conda-forge/staged-recipes/pull/23318#discussion_r1259924034).
- The singleton_decorator project is clearly abandoned (last commit seven years ago) and [previous attempts to add a license have been left unsanswered](https://github.com/Kemaweyan/singleton_decorator/issues/6).

One solution would be to fork singleton_decorator and release it under a proper license. One would then have to update jupyterlab-s3-browser to use the new dependency. However, given that jupyterlab-s3-browser itself is also somewhat abandoned (last commit two years ago), it might be easier to fork from there and create a new s3-browser library altogether.

**Pull request closed without merging**